README.md

```diff
@@ -1,2 +1,179 @@
+                        </dependency>
+                    </dependencies>
+                </plugin>
+```
+
+### Example result output
+When executing your next Maven build, Baton will execute as part of the `initialize` build lifecycle and output similar
+to the following will result:
+
+```bash
+[INFO] --- baton:0.1.0:baton-migrate (default) @ toml-file-update ---
+[INFO] Loading migrations from: jar:file:/Users/dfh/.m2/repository/org/technologybrewery/baton/example/example-migration-configuration/1.0.0/example-migration-configuration-1.0.0-SNAPSHOT.jar!/migrations.json
+[INFO] Found 1 migrations
+[INFO] Migrations Processed: 1, Successfully Migrated Files: 1, Unsuccessfully Migrated Files: 0
+```
+
+## Configuration
+All Baton configurations may be set either via the `baton-maven-plugin`'s `<configuration>` definition, Maven POM 
+properties, or `-D` on the line and follow a consistent naming pattern for the different configuration approaches.  For 
+setting configurations via POM properties or `-D` on the command line, all configuration keys may be prepended with 
+`baton.`.  For example, `migrationsConfigurationFile` controls the file name that Baton will use to find migrations 
+and may be configured using the following approaches:
+
+1. Plugin `<configuration>`
+
+```xml
+	<plugin>
+		<groupId>org.technologybrewery.baton</groupId>
+		<artifactId>baton-maven-plugin</artifactId>
+		<extensions>true</extensions>
+		<configuration>
+			<migrationsConfigurationFile>alternative-migrations.json</migrationsConfigurationFile>
+		</configuration>
+	</plugin>
+```
+
+2. `-D` via command line
+
+```shell
+mvn clean install -DmigrationsConfigurationFile=alternative-migrations.json
+```
+
+3. POM properties
+
+```xml
+	<properties>
+		<migrationsConfigurationFile>alternative-migrations.json</migrationsConfigurationFile>
+	</properties>
+```
+
+**NOTE:** The above list's order reflects the precedence in which configurations will be applied.  For example, 
+configuration values that are specified in the plugin's `<configuration>` definition will always take precedence, while 
+system properties via the command line (`-D`) will take precedence over `<properties>` definitions.
+
+### baseDirectory
+The desired base directory to use when Baton looks for files on which to perform migrations.  By default, only `pom.xml`
+and `*.toml` from the base directory will be included.
+
+Default: `${project.basedir}`
+
+### sourceDirectory
+The desired source directory to include when Baton looks for files on which to perform migrations.
+
+Default: `${project.basedir}/src`
+
+### testDirectory
+The desired test directory to include, if desired, when Baton looks for files on which to perform migrations.
+
+Default: None
+
+### fileSets
+A [standard Maven fileSets block](https://maven.apache.org/shared/file-management/examples/mojo.html) that can be used 
+to include any sets of file you desire.  If used, the default fileSets information will not be added.
```

Comment:
Default fileSets information meaning `pom.xml` and `*.toml` files?  If so, this might read better with the `By default...` sentence from `baseDirectory` moved here in lieu of this sentence.

Reply:
Updated to the following to try and improve clarity:

>  If used, no additional defaulting of fileSet values will be performed by Baton and *only* these filesets will be used for the plugin's execution.